src/coreservices.cpp

```diff
@@ -123,7 +124,137 @@ CoreServices::CoreServices(const CmdlineArgs& args, QApplication* pApp)
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting RecordingManager";
+    CLEAR_AND_CHECK_DELETED(m_pRecordingManager);
+
+#ifdef __BROADCAST__
+    // BroadcastManager depends on config, engine
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting BroadcastManager";
+    CLEAR_AND_CHECK_DELETED(m_pBroadcastManager);
+#endif
+
+    // EngineMaster depends on Config and m_pEffectsManager.
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting EngineMaster";
+    CLEAR_AND_CHECK_DELETED(m_pEngine);
+
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting EffectsManager";
+    CLEAR_AND_CHECK_DELETED(m_pEffectsManager);
+
+    // Delete the track collections after all internal track pointers
+    // in other components have been released by deleting those components
+    // beforehand!
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "detaching all track collections";
+    CLEAR_AND_CHECK_DELETED(m_pTrackCollectionManager);
+
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "closing database connection(s)";
+    m_pDbConnectionPool->destroyThreadLocalConnection();
+    m_pDbConnectionPool.reset(); // should drop the last reference
+
+    // HACK: Save config again. We saved it once before doing some dangerous
+    // stuff. We only really want to save it here, but the first one was just
+    // a precaution. The earlier one can be removed when stuff is more stable
+    // at exit.
+    m_pSettingsManager->save();
+
+    m_pTouchShift.reset();
+
+    m_pControlIndicatorTimer.reset();
+
+    // Check for leaked ControlObjects and give warnings.
+    {
+        const QList<QSharedPointer<ControlDoublePrivate>> leakedControls =
+                ControlDoublePrivate::takeAllInstances();
+        if (!leakedControls.isEmpty()) {
+            qWarning()
+                    << "The following"
+                    << leakedControls.size()
+                    << "controls were leaked:";
+            for (auto pCDP : leakedControls) {
+                ConfigKey key = pCDP->getKey();
+                qWarning() << key.group << key.item << pCDP->getCreatorCO();
+                // Deleting leaked objects helps to satisfy valgrind.
+                // These delete calls could cause crashes if a destructor for a control
+                // we thought was leaked is triggered after this one exits.
+                // So, only delete so if developer mode is on.
+                if (CmdlineArgs::Instance().getDeveloper()) {
+                    pCDP->deleteCreatorCO();
+                }
+            }
+            DEBUG_ASSERT(!"Controls were leaked!");
+        }
+        // Finally drop all shared pointers by exiting this scope
+    }
+
+    Sandbox::shutdown();
+
+    qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting SettingsManager";
+    m_pSettingsManager.reset();
+
+    CLEAR_AND_CHECK_DELETED(m_pKeyboardEventFilter);
```

Comment:
This and below is initialized in the C-Tor and skipped if initialize() has not been called.   
Maybe it was a good idea to implement a counter part to initialize() to make this obvious.